BootCampV2/Homeworks/SecondWeek/ArrayExtension.cs

```diff
@@ -0,0 +1,149 @@
+using System;
+
+namespace BootCampV2
+{
+    public class ArrayExtension
+    {
+        /// <summary>
+        /// Use the bubble sort algorithm to sort an array
+        /// <para> return: sorted Array </para> 
+        /// </summary>
+        public static int[] BubbleSort(int[] array)
+        {
+            for (int i = 0; i < array.Length; i++)
+            {
+                bool swap = false;
+                for (int m = 0; m < (array.Length - i) - 1; m++)
+                {
+                    if (array[m] > array[m + 1])
+                    {
+                        array = SwapElements(array, m, m + 1);
+                        swap = true;
+                    }
+                }
+                if (!swap)
+                    break;
+            }
+            return array;
```

Comment:
Typically I would expect a sort method to either modify the array I pass in (in which case the return type is void) or make a copy of the array and return the sorted array as a copy.  Here you are returning an array (so I might think you are making a copy) but you modify the array passed in.  This would be a little confusing to a user.